lib/sudo-exec/src/lib.rs

```diff
@@ -30,10 +31,23 @@ pub fn run_command(ctx: Context<'_>, env: Environment) -> io::Result<ExitStatus>
     let mut command = Command::new(ctx.command.command);
     // reset env and set filtered environment
     command.args(ctx.command.arguments).env_clear().envs(env);
+    if let Some(path) = ctx.chdir {
+        // change current directory, if requested
+        command.current_dir(path);
+    } else if ctx.login {
+        // change current directory, if `-i` is being used and the home directory exists.
+        let path = Path::new(&ctx.target_user.home);
+        if path.exists() {
+            command.current_dir(path);
+        } else {
```

Comment:
I think it's better from now on to use the `sudo_common::Error` and the standard Result type to return errors upwards instead of handling them themselves; we have a diagnostic handler there which also does things like setting the exit code of the program. If changing the directory failed that should just mean execution failed, right?

Reply:
> If changing the directory failed that should just mean execution failed, right?
Sort of, what ogsudo does when the `-i` flag is set is:
- Change the current directory to the user home if it exists.
- Do not change the current directory if the user home does not exist and print a warning to `stderr` about it.

In the first scenario, if there is some kind of issue changing the current directory, executing the command will fail with an `io::Error` that will be propagated as you mentioned.

In the second scenario, there's no actual error as the execution just continues without changing the current directory.

Reply:
Right. CC #21 in that case :)